feat(sitemap): add configurable XML namespaces support (#12250)
withastro/docs: src/content/docs/en/guides/integrations-guide/sitemap.mdx

The given configuration will generate sitemap files at `https://example.com/astronomy-sitemap-0.xml` and `https://example.com/astronomy-sitemap-index.xml`.

### `namespaces`

**Type:** `{ news?: boolean; xhtml?: boolean; image?: boolean; video?: boolean; }`  
**Default:** `{ news: true, xhtml: true, image: true, video: true }`

Controls which XML namespaces are included in the sitemap generation.

By default, all namespaces default to `true` and you must explicitly set each one you want to change to `false`.

```js title="astro.config.mjs" ins={8-14}
import { defineConfig } from 'astro/config';
import sitemap from '@astrojs/sitemap';

export default defineConfig({
  site: 'https://example.com',
  integrations: [
    sitemap({
      // Only include image and video namespaces
      namespaces: {
        news: false,
        xhtml: false, 
        // image and video will be true by default
      }
    })
  ]
});

> 

Reviewer: ``````suggestion
```

``````

> export default defineConfig({
>   site: 'https://example.com',
>   integrations: [
>     sitemap({
>       // Only include image and video namespaces
>       namespaces: {
>         news: false,
>         xhtml: false, 
>         // image and video will be true by default
>       },
>     }),
>   ],
> });

Reviewer 2: removed trailing commas
```suggestion
export default defineConfig({
  site: 'https://example.com',
  integrations: [
    sitemap({
      // Only include image and video namespaces
      namespaces: {
        news: false,
        xhtml: false, 
        // image and video will be true by default
      }
    })
  ]
});
```

The given configuration will generate sitemap files at `https://example.com/astronomy-sitemap-0.xml` and `https://example.com/astronomy-sitemap-index.xml`.

### `namespaces`

**Type:** `{ news?: boolean; xhtml?: boolean; image?: boolean; video?: boolean; }`  
**Default:** `{ news: true, xhtml: true, image: true, video: true }`

> Controls which XML namespaces are included in the sitemap generation.

Author: Why someone would want to exclude namespaces:
The main practical reasons are:
- Excluding unused namespaces reduces XML bloat
- Smaller sitemap files are faster to parse by search engines
- Some SEO analysis tools prefer minimal sitemaps without unused declarations
- Particularly relevant for sites with very large sitemaps

Regarding the 4 namespace options:
These are the standard XML namespaces that Astro includes by default in sitemaps. 

- xmlns:news - Google News sitemaps
- xmlns:xhtml - Multilingual/alternate language links
- xmlns:image - Image sitemaps
- xmlns:video - Video sitemaps

You're right that we could potentially add more namespaces in the future, and separate subsections might be worth considering as the list grows.

Suggested documentation revision:

### `namespaces`

<p>

**Type:** `{ news?: boolean; xhtml?: boolean; image?: boolean; video?: boolean; }` <br />
**Default:** `{ news: true, xhtml: true, image: true, video: true }` <br />
<Since v="3.6.0" pkg="@astrojs/sitemap" />
</p>

An object of XML namespaces to exclude from the generated sitemap.

Excluding unused namespaces can help create cleaner, more focused sitemaps that are faster for search engines to parse and use less bandwidth. For example, if your site doesn't have news content, videos, or multiple languages, you can exclude those namespaces to reduce XML bloat.

By default, all configurable namespaces  (`news`, `xhtml`, `image`, and `video`) are included in your generated sitemap XML. To exclude a namespace from your sitemap generation, add a `namespaces` configuration object and set individual options to `false`:

```js title="astro.config.mjs" ins={8-14}
import { defineConfig } from 'astro/config';
import sitemap from '@astrojs/sitemap';

export default defineConfig({
  site: 'https://example.com',
  integrations: [
    sitemap({
      // Only include image and video namespaces
      namespaces: {
        news: false,
        xhtml: false, 
        // image and video will be true by default
      }
    })
  ]
});
```

The given configuration will generate sitemap files at `https://example.com/astronomy-sitemap-0.xml` and `https://example.com/astronomy-sitemap-index.xml`.

### `namespaces`

<p>

**Type:** `{ news?: boolean; xhtml?: boolean; image?: boolean; video?: boolean; }` <br />
**Default:** `{ news: true, xhtml: true, image: true, video: true }` <br />
<Since v="3.6.0" pkg="@astrojs/sitemap" />
</p>

An object of XML namespaces to exclude from the generated sitemap.

> Excluding unused namespaces can help create cleaner, more focused sitemaps that are faster for search engines to parse and use less bandwidth. For example, if your site doesn't have news content, videos, or multiple languages, you can exclude those namespaces to reduce XML bloat.

Reviewer: ```suggestion
Excluding unused namespaces can help create more focused sitemaps that are faster for search engines to parse and use less bandwidth. For example, if your site doesn't have news content, videos, or multiple languages, you can exclude those namespaces to reduce XML bloat.
```
I love this paragraph, thank you so much for adding it! I think we can stick to just "more focused" to keep this... more focused! 😂

> By default, all configurable namespaces  (`news`, `xhtml`, `image`, and `video`) are included in your generated sitemap XML. To exclude a namespace from your sitemap generation, add a `namespaces` configuration object and set individual options to `false`:

Reviewer: ```suggestion
By default, all configurable namespaces (`news`, `xhtml`, `image`, and `video`) are included in your generated sitemap XML. To exclude one or more of these namespaces from your sitemap generation, add a `namespaces` configuration object and set individual options to `false`:
```

```js title="astro.config.mjs" ins={8-14}
import { defineConfig } from 'astro/config';
import sitemap from '@astrojs/sitemap';

export default defineConfig({
  site: 'https://example.com',
  integrations: [
    sitemap({

>       // Only include image and video namespaces
>       namespaces: {
>         news: false,
>         xhtml: false, 
>         // image and video will be true by default

Reviewer: ```suggestion
      namespaces: {
        news: false,
        xhtml: false, 
```
I think this code snippet works even without the comments! I had originally thought about keeping one or the other, but I'm not sure it's needed.
